File: lib/src/main/java/com/healthassist/service/HospitalService.java
package com.healthassist.service;

import java.util.List;
import java.util.Optional;

import com.healthassist.entities.Doctor;
import com.healthassist.entities.DoctorSpeciality;
import com.healthassist.entities.Hospital;
import com.healthassist.entities.TreatmentHistory;
import com.healthassist.exception.NoBedsFoundException;


public interface HospitalService {
	 public void addHospitalInfo(Hospital hospital);
	 public List<Hospital> findHospital();
	 public Doctor addDoctor(Doctor doctor);
	 public List<Doctor> getDoctor(Doctor doctor);
	 public Optional<Hospital> updateDoctor(int doctorId);
	 public void deleteDoctor(int doctorId);
	 public Integer viewNoOfBeds();
	 public Hospital modifyNoOfBeds(int noOfBeds, int hospitalId)  throws NoBedsFoundException;
	 public List<DoctorSpeciality> viewDoctoryBySpeciality(String doctorSpeciality);
	 public void addTreatmentHistory(TreatmentHistory treatmentHistory);
	 public TreatmentHistory updateTraetmentHistory(TreatmentHistory treatmenthistory);
	 
}

File: src/hospital/service/HospitalService.js
import axios from 'axios';
class HospitalService{
addDoctor(doctor){
 return axios.post('http://localhost:9090/LIFELINE-api/LIFELINE/addDoctorInHospital', doctor);
}
getDoctor(){
    return axios.get('http://localhost:9090/LIFELINE-api/LIFELINE/getDoctorFromHospital');
}
deleteDoctor(doctorId){
    return axios.delete(`http://localhost:9090/LIFELINE-api/LIFELINE/deleteDoctorInHospital/${doctorId}`);
}
updateDoctor(doctor){
    return axios.put('http://localhost:9090/LIFELINE-api/LIFELINE/updateDoctorInHospital',doctor);
}
findDoctorById(doctorId){
    return axios.get(`http://localhost:9090/LIFELINE-api/LIFELINE/getDoctorById//${doctorId}`);
}
addHospital(hospital){
    return axios.post('http://localhost:9090/LIFELINE-api/LIFELINE/addHospital', hospital);
}
getHospital(){
    return axios.get('http://localhost:9090/LIFELINE-api/LIFELINE/getHospital');
}
viewNoOfBeds(hospital){
    return axios.get('http://localhost:9090/LIFELINE-api/LIFELINE/viewNoOfBeds', hospital);
}
addTreatmentHistory(treatmentHistory){
    return axios.post('http://localhost:9090/LIFELINE-api/LIFELINE/addTreatmentHistory', treatmentHistory);
}
getTreatmentHistory(){
    return axios.get(`http://localhost:9090/LIFELINE-api/LIFELINE/getTeatmentHistory`);
}
findDoctorBySpeciality(doctor){
    return axios.get('http://localhost:9090/LIFELINE-api/LIFELINE/findDoctorbySpeciality', doctor);
}
findHospitalByid(hospitalId){
    return axios.get(`http://localhost:9090/LIFELINE-api/LIFELINE/getHospitalById/${hospitalId}`);
}
updateNoOfBeds(hospital){
    return axios.put('http://localhost:9090/LIFELINE-api/LIFELINE/updateNoOfBedsInHospital', hospital);
}
deleteHospital(hospitalId){
    return axios.delete(`http://localhost:9090/LIFELINE-api/LIFELINE/deleteHospital/${hospitalId}`);
}
}
export default HospitalService;